cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -660,6 +667,30 @@ def _select_ds_for_volume(self, volume, host=None, create_params=None):
 
         return (host_ref, resource_pool, folder, summary)
 
+    @property
+    def service_locator_info(self):
+        """Returns information needed to build a ServiceLocator spec."""
+        # vCenter URL
+        host = self.configuration.vmware_host_ip
+        port = self.configuration.vmware_host_port
+        url = "https://" + host
+        if port:
+            url += ":" + str(port)
+        # ssl thumbprint
+        cert = ssl.get_server_certificate((host, port or 443))
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM,
+                                               cert)
+        return {
+            'url': url,
+            'ssl_thumbprint': x509.digest("sha1"),
+            'instance_uuid':
+                self.session.vim.service_content.about.instanceUuid,
+            'credential': {
```

Comment:
Wouldn't `credentials` with an `s` be a better name?

Reply:
Maybe it is. 
However `credential` follows the naming from the VMware's `vim.ServiceLocator`.

Reply:
Ah, that's where it's coming from. Thanks.